swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +53,49 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static void checkClassEndpoints(Map<String, SwaggerEndpoint> existingPaths, Map<String, SwaggerEndpoint> classPaths) {
+		if (!existingPaths.isEmpty()) {
+			for(Map.Entry<String, SwaggerEndpoint> pathEntry: existingPaths.entrySet()) {
+				for(Map.Entry<String, SwaggerEndpoint> classEntry: classPaths.entrySet()) {

```

Comment:
Instead of these nested for loops:

We don't need the entry set, just the keys:
`for(String key : existingPaths.keySet())`

Next, we don't need to loop over the entries of a map looking for matching keys. Maps already provide the tools necessary. 
```
if(classPaths.hasKey(key)) {
    existingPaths.get(key).addEndpoint(classPaths.get(key);
}
```